File: practice/algorithms/bit_manipulation/maximizing_xor/MaximizingXor.java
import java.util.*;

public class MaximizingXor {

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int l = stdin.nextInt();
        int r = stdin.nextInt();
        int max = 0;
        for(int i = l; i <= r; i++) {
            for(int j = l; j <= r; j++) {
                int xor = i ^ j;
                if(xor > max) {
                    max = xor;
                }
            }
        }
        System.out.println(max);
        stdin.close();
    }
}

File: practice/algorithms/bit_manipulation/maximizing_xor/MaximizingXor.js
function processData(input) {
    var l = parseInt(input.split("\n")[0]);
    var r = parseInt(input.split("\n")[1]);
    var max = 0;
    for(var i = l; i <= r; i++) {
        for(var j = l; j <= r; j++) {
            var xor = i ^ j;
            if(xor > max) {
                max = xor;
            }
        }
    }
    console.log(max);
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
    processData(_input);
});